scripts/extensions/sams/src/apps/samsAttachmentsWidget.tsx

```diff
@@ -108,10 +108,19 @@ class SamsAttachmentsWidgetComponent extends React.PureComponent<IProps> {
     }
 
     showSelectAssetModal() {
+        let attachmentMediaIds: Array<string> = [];
+
+        this.props.attachments.forEach((attachment) => {
+            let media = attachment.media as IMedia;
+
+            attachmentMediaIds = attachmentMediaIds.concat(media._id);
+        });
+
         this.props.setAssetSearchParams({
             sizeTo: superdeskApi.instance.config.attachments_max_size / 1048576, // bytes -> MB
             states: [ASSET_STATE.PUBLIC, ASSET_STATE.INTERNAL],
             setIds: this.props.activeSets.map((set) => set._id),
+            assetIds: attachmentMediaIds,
```

Comment:
This can be simplified into a single line assignment:
```ts
excludeAssetIds: this.props.attachments.map((attachment) => (
    typeof attachment.media === 'string' ?
        attachment.media :
        attachment.media._id
)),
```